src/main/java/io/boomerang/eventing/nats/jetstream/PubTransmitter.java

```
@@ -80,32 +87,114 @@ public void publish(String subject, String message)
+    // break if queue is empty or if connection fails again
+    
+    while (true) {
+      synchronized (this) {
+        try {
+          System.out.println("Resending failed message");
+          PublishAck publishAck =
+              connectionPrimer.getActiveConnection().jetStream().publish(failedMessages.peek());
+          System.out.println("PublishAck:");
+          printObject(publishAck);
+          if (publishAck != null) {
+            System.out.println("-------------------------------------------------");
+            System.out.println("Queue size before: " + failedMessages.size());
+            printObject(failedMessages);
+            
+            System.out.println("Removing published message");
+            failedMessages.poll();
+            
+            System.out.println("Queue size after: " + failedMessages.size());
+            printObject(failedMessages);
+            System.out.println("-------------------------------------------------");
+          }
+        } catch (IOException | JetStreamApiException e) {
+          e.printStackTrace();
+        }
+
+        if (failedMessages == null || connectionPrimer.getActiveConnection() == null) {
+          break;
+        }
+      }
+    }
+  }
 
-    logger.debug("Message published to the stream! " + publishAck);
+  // TODO remove this after testing, also remove dependency from pom.xml
+  private void printObject(Object object) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    try {
+      System.out.println(objectMapper.writeValueAsString(object));
+    } catch (JsonProcessingException e) {
+      // log an error
+    }
+  }
+
+  @Override
+  public void connectionUpdated(ConnectionPrimer connectionPrimer) {
+    // TODO don't retry messages if failed stream
+    Connection connection = connectionPrimer.getActiveConnection();
+    if (connection != null) {
+      System.out.println("Connection active");
+      try {
+        getJetStream(connection);
```

If this fails due to no Stream being present on the server (based on the provided configuration) and `automaticallyCreateStream` is set to `false`, there is no reason to re-publish the messages, since these will fail anyway.
The code here should fail silently (for now) and you must clear out the queue.